subsys/net/openthread/rpc/server/ot_rpc_srp_client.c

```diff
@@ -0,0 +1,812 @@
+		ot_rpc_report_decoding_error(OT_RPC_CMD_SRP_CLIENT_SET_TTL);
+		return;
+	}
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	otSrpClientSetTtl(openthread_get_default_instance(), ttl);
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_void(group);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_set_ttl, OT_RPC_CMD_SRP_CLIENT_SET_TTL,
+			 ot_rpc_cmd_srp_client_set_ttl, NULL);
+
+static void ot_rpc_cmd_srp_client_start(const struct nrf_rpc_group *group,
+					struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	otSockAddr sockaddr;
+	otError error;
+
+	ot_rpc_decode_sockaddr(ctx, &sockaddr);
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		ot_rpc_report_decoding_error(OT_RPC_CMD_SRP_CLIENT_START);
+		return;
+	}
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	error = otSrpClientStart(openthread_get_default_instance(), &sockaddr);
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_uint(group, error);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_start, OT_RPC_CMD_SRP_CLIENT_START,
+			 ot_rpc_cmd_srp_client_start, NULL);
+
+static void ot_rpc_cmd_srp_client_stop(const struct nrf_rpc_group *group,
+					struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	otSrpClientStop(openthread_get_default_instance());
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_void(group);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_stop, OT_RPC_CMD_SRP_CLIENT_STOP,
+			 ot_rpc_cmd_srp_client_stop, NULL);
+
+static void ot_rpc_cmd_srp_client_is_running(const struct nrf_rpc_group *group,
+					     struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	bool running;
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	running = otSrpClientIsRunning(openthread_get_default_instance());
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	nrf_rpc_rsp_send_bool(group, running);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(ot_group, ot_rpc_cmd_srp_client_is_running,
+			 OT_RPC_CMD_SRP_CLIENT_IS_RUNNING, ot_rpc_cmd_srp_client_is_running, NULL);
+
+static void ot_rpc_cmd_srp_client_get_server_address(const struct nrf_rpc_group *group,
+						      struct nrf_rpc_cbor_ctx *ctx,
+						      void *handler_data)
+{
+	struct nrf_rpc_cbor_ctx rsp_ctx;
+	const otSockAddr *sockaddr;
+
+	nrf_rpc_cbor_decoding_done(group, ctx);
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, rsp_ctx, OT_IP6_ADDRESS_SIZE + sizeof(uint32_t) + 2);
+
+	openthread_api_mutex_lock(openthread_get_default_context());
+	sockaddr = otSrpClientGetServerAddress(openthread_get_default_instance());
+	openthread_api_mutex_unlock(openthread_get_default_context());
+
+	ot_rpc_encode_sockaddr(&rsp_ctx, sockaddr);
```

Comment:
You should probably do it before unlocking the mutex or save a copy instead of a pointer.